### PR TITLE
Cleanup-1 [ODE Tutorial update, inti fix, termination, minor dependency fix]

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Shiva-sankaran — COMMENTED** [review]
* The .py changes seem harmless to me. (LGTM)
* Not sure about the .gitignore and the notebook changes.

File: deepchem/models/torch_models/flows.py [hunk 1]
```diff
@@ -863,7 +859,8 @@ def fit(  # type: ignore
                               lr=learning_rate,
                               weight_decay=weight_decay)
         nfm = self.nfm
-        for epoch in tqdm(range(epochs)):
```
Comment (rbharath) [inline deepchem/models/torch_models/flows.py:866]: Please remove these changes

File: .gitignore [hunk 2]
```diff
@@ -107,5 +107,6 @@ datasets/qm8.csv
 env.yml
 .python-version
 **/*.ipynb_checkpoints/
+.virtual_documents
```
Comment (rbharath) [inline .gitignore:110]: Please do some research on what this .virtual_documents

File: deepchem/utils/differentiation_utils/optimize/minimizer.py [hunk 3]
```diff
@@ -61,6 +62,9 @@ def gd(
         The absolute tolerance of the norm of the input ``x``.
     x_rtol: float or None
         The relative tolerance of the norm of the input ``x``.
+    terminate: bool (default True)
+        Wether to use termiation condition or, keep on running the
```
Comment (rbharath) [inline deepchem/utils/differentiation_utils/optimize/minimizer.py:66]: Typo "Whether", "Termination"

File: deepchem/utils/differentiation_utils/optimize/minimizer.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ def gd(
         x_rtol: float = 1e-8,
         # misc parameters
         verbose=False,
+        terminate=True,
```
Comment (rbharath) [inline deepchem/utils/differentiation_utils/optimize/minimizer.py:21]: You need a unit test for terminate

File: examples/tutorials/Ordinary_Differential_Equation_Solving_using_deepchem.ipynb [hunk 5]
```diff
@@ -109,7 +109,7 @@
   },
```
Comment (rbharath) [inline examples/tutorials/Ordinary_Differential_Equation_Solving_using_deepchem.ipynb:109]: <p>Use US spelling "minimizer" as default to match style elsewhere.</p><p> </p><p>Also add some scientific discussion of how the parameter estimation works</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4103/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (rbharath) [inline examples/tutorials/Ordinary_Differential_Equation_Solving_using_deepchem.ipynb:109]: <p>Add some discussion of the results; how close do we get to the true parameters?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4103/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

File: .gitignore (hunk 2 above)
Comment (GreatRSingh) [reply .gitignore:110]: It temporarily stores jupyter notebooks when using an lsp server with jupyter labs.

our backend Jupyter server extension creates a temporary file on the file system (by default in the .virtual_documents directory). [1]

References
----
[1].. https://jupyter.org/enhancement-proposals/72-language-server-protocol/language-server-protocol.html

**Pull request closed without merging**